### ISSUE TITLE
playerStateChanged triggers callback multiple times

### ISSUE THREAD

**Issue by MrBuggySan** [cjam/react-native-spotify-remote#66]
https://github.com/cjam/react-native-spotify-remote/blob/c0a81d7fb4dae4686b713268aa51471188729562/example/AppContext.tsx#L73 

`remote.on("playerStateChanged", this.onPlayerStateChanged);`

When playerStateChanged is triggered, the callback is called multiple times. I observe this with the example app and also with my app using this library. Any clue as to what might be causing this?

**Comment by cjam** [comment]
What do you mean by multiple times? Which platform?

**Comment by MrBuggySan** [comment]
When Spotify player state changes (ie song changes) onPlayerStateChanged method gets called multiple times instead of just once. I observed this on Android, I haven't tested with iOS

**Comment by uptotec** [comment]
@MrBuggySan I had the same problem in android also and I found the solution is to use the callback pattern reference [here](https://reactjs.org/docs/hooks-reference.html#usecallback) .
this is the pattern
```javascript
const SpotifyCallback = React.useCallback(async (currentPlayerState) => {
//write you code here
  }, []);

  React.useEffect(() => {
    SpotifyRemote.on('playerStateChanged', SpotifyCallback);
    return () => {
      SpotifyRemote.removeListener('playerStateChanged', SpotifyCallback);
    };
  }, [SpotifyCallback]);
```

**Comment by cjam** [comment]
@uptotec yes you definitely want to return a cleanup function if subscribing using useEffect. I probably need to update the example app in this repo.
